### ISSUE TITLE
💻 Public Adventures interface ideas

### ISSUE THREAD

**Issue by MarleenGilsing** [hedyorg/hedy#5475]
I was making a public adventure on this page and I have some ideas for improvements:

![image](https://github.com/hedyorg/hedy/assets/80678586/3859b311-051f-431f-8843-75b447ecc2cf)

'Available in' doesn't specify that the teacher is meant to choose the level. So can we change it to: 'Available in level' or just 'level' if the former is too long. This will make it clear for the teachers that we want them to choose the level. 

Maybe we can make some tags, instead of letting the teachers create their own tags, because at the moment their are a lot of nonsense tags. This defeats the purpose of a tag. 

In the previous verision the teacher could see how to use the `` to make the text look like `print` in the example, now `print` already looks like that in the example, so teachers won't know how to do that. 

Lastly, and to me most importantly, I'm missing a 'submit' button. When I made the adventure I clicked 'preview' at least 5 times to make sure I didn't miss a 'submit' or 'save' button. It feels very weird to just click 'go back' when you're done and just hope your work has been saved. Can we add a 'submit' or 'save adventure' button in the row with the 'remove' and 'preview' button?

**Comment by hasan-sh** [comment]
> In the previous verision the teacher could see how to use the `` to make the text look like `print` in the example, now `print` already looks like that in the example, so teachers won't know how to do that.

So if you add the backticks, you'd make a keyword which is what to expect right? Not sure if you meant something else!



> Lastly, and to me most importantly, I'm missing a 'submit' button. When I made the adventure I clicked 'preview' at least 5 times to make sure I didn't miss a 'submit' or 'save' button. It feels very weird to just click 'go back' when you're done and just hope your work has been saved. Can we add a 'submit' or 'save adventure' button in the row with the 'remove' and 'preview' button?

We removed that button because everything now is auto-saved. So, teachers don't need to save it manually.

**Comment by MarleenGilsing** [comment]
> > In the previous verision the teacher could see how to use the `` to make the text look like `print` in the example, now `print` already looks like that in the example, so teachers won't know how to do that.
> 
> So if you add the backticks, you'd make a keyword which is what to expect right? Not sure if you meant something else!

I meant teachers probably won't know how to accomplish that look with the backticks. It isn't explained anywhere how to do that, you just see the correct endresult, but not how to get there, if you understand what i mean. 


> > Lastly, and to me most importantly, I'm missing a 'submit' button. When I made the adventure I clicked 'preview' at least 5 times to make sure I didn't miss a 'submit' or 'save' button. It feels very weird to just click 'go back' when you're done and just hope your work has been saved. Can we add a 'submit' or 'save adventure' button in the row with the 'remove' and 'preview' button?
> 
> We removed that button because everything now is auto-saved. So, teachers don't need to save it manually.

Yes, autosafe is great! But if the only way to 'get out' after finishing the adventure is to go back, it feels counterintuitive. So a continue button, or a submit button that takes you to the dashboard would still make sense to me.

**Comment by hasan-sh** [comment]
> I meant teachers probably won't know how to accomplish that look with the backticks. It isn't explained anywhere how to do that, you just see the correct endresult, but not how to get there, if you understand what i mean.

Yes i understand. In https://github.com/hedyorg/hedy/issues/5356, the explanation box should indeed explain that if you click on the "code" icon, you'd create a keyword. And if you click on the "block code" icon, you'd create a block of 1 or more code lines.



> Yes, autosafe is great! But if the only way to 'get out' after finishing the adventure is to go back, it feels counterintuitive. So a continue button, or a submit button that takes you to the dashboard would still make sense to me.

I totally agree! Perhaps we could change the text of "go back" to something like "Teacher dashboard".

**Comment by Annelein** [comment]
@Felienne @hasan-sh is working on the Public Adventure page at the moment so I thought it would make sense for him to include this issue.

**Comment by hasan-sh** [comment]
This would mean I need to know what exactly you'd want to improve here. So, it'd be great if @Felienne could reply to the discussion above!

**Comment by Felienne** [comment]
> This would mean I need to know what exactly you'd want to improve here. So, it'd be great if @Felienne could reply to the discussion above!
> 
> Also, this should be implemented after #5431, since the latter brings many changes!

Hi Hasan!

As far as I can see all Marleen's points have been answered by you, so I am not sure what input is needed? Let me know what you need and I can read along.

**Comment by hasan-sh** [comment]
> 'Available in' doesn't specify that the teacher is meant to choose the level. So can we change it to: 'Available in level' or just 'level' if the former is too long. This will make it clear for the teachers that we want them to choose the level.

Changed to level. 


> In the previous verision the teacher could see how to use the `` to make the text look like `print` in the example, now `print` already looks like that in the example, so teachers won't know how to do that.

These can be explained in https://github.com/hedyorg/hedy/blob/main/messages.pot#L194-L198 
the first si about the inline code (i.e., `print`). Would you want to explain it as you see fit?



> Lastly, and to me most importantly, I'm missing a 'submit' button. When I made the adventure I clicked 'preview' at least 5 times to make sure I didn't miss a 'submit' or 'save' button. It feels very weird to just click 'go back' when you're done and just hope your work has been saved. Can we add a 'submit' or 'save adventure' button in the row with the 'remove' and 'preview' button?

Will add a submit button!


> Maybe we can make some tags, instead of letting the teachers create their own tags, because at the moment their are a lot of nonsense tags. This defeats the purpose of a tag.

This needs to be discussed and decided; so in another PR?
